File: ruby/gmsec4/dox/Field.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.
#


module Libgmsec_ruby #:nodoc: don't document this

	# This is the abstract base Field class for all field types.
	#
	class Field

        ##
        # Possible field types for a Field object.
        #
        class FieldType < ActiveRecord::Base
            ##
            # ==== Enumerated values
            #
            # * +CHAR_TYPE+    - Field containing an ASCII character value.
            # * +BOOL_TYPE+    - Field containing a boolean value.
            # * +I16_TYPE+     - Field containing a signed 16-bit value.
            # * +U16_TYPE+     - Field containing an unsigned 16-bit value.
            # * +I32_TYPE+     - Field containing a signed 32-bit value.
            # * +U32_TYPE+     - Field containing an unsigned 32-bit value.
            # * +F32_TYPE+     - Field containing a 32-bit floating point value.
            # * +F64_TYPE+     - Field containing a 64-bit floating point value.
            # * +STRING_TYPE+  - Field containing a string.
            # * +BIN_TYPE+     - Field containing raw binary data.
            # * +I8_TYPE+      - Field containing a signed 8-bit value.
            # * +U8_TYPE+      - Field containing an unsigned 8-bit value.
            # * +I64_TYPE+     - Field containing a signed 64-bit value.
            # * +U64_TYPE+     - Field containing an unsigned 64-bit value.
            #
            def FieldType
            end
        end


		# Returns the name associated with the field.
		#
		# ==== Returns
		#
		# A string containing the field name.
		#
		def get_field_name
		end


		# Sets the field name with the given string.
		#
		# ==== Attributes
		#
		# * +name+ - The name to assign to the field.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the field name is nil or contains an empty string.
		#
		def set_field_name(name)
		end


		# Returns the type of the field.
		#
		# ==== Returns
		#
		# A FieldType enumerated value.
		#
		def get_field_type
		end


		# Used to indicate if the field is a header field (see C2MS definition).
		#
		# ==== Attributes
		#
		# * +flag+ - Use true to indicate a header field; false otherwise.
		#
		def set_header(flag)
		end


		# Can be used to query if the field is a header field (see C2MS definition).
		#
		# ==== Returns
		#
		# Returns true if the field represents a header field; false otherwise.
		#
		def is_header()
		end


		# Converts the field info into an XML string representation.
		#
		# ==== Returns
		#
		# An XML string.
		#
		def to_xml
		end


		# Converts the field info into a JSON string representation.
		#
		# ==== Returns
		#
		# A JSON string.
		#
		def to_json
		end


		# Attempts to convert the field value into a 64-bit integer representation.
		#
		# ==== Returns
		#
		# Return the field value as an integer.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the field cannot be successfully converted to an integer.
		#
		def get_integer_value
		end


		# Attempts to convert the field value into a 64-bit unsigned integer representation.
		#
		# ==== Returns
		#
		# Return the field value as an unsigned integer.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the field cannot be successfully converted to an unsigned integer.
		#
		def get_unsigned_integer_value
		end


		# Attempts to convert the field value into a 64-bit floating point representation.
		#
		# ==== Returns
		#
		# Return the field value as a floating point.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the field cannot be successfully converted to a floating point.
		#
		def get_double_value
		end


		# Attempts to convert the field value into string representation.
		#
		# ==== Returns
		#
		# Return the field value as a string.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the field cannot be successfully converted to a string.
		#
		def get_string_value
		end


		# Creates an exact clone of the existing field object.
		#
		# ==== Returns
		# A clone of a Field object.
		#
		def clone
		end
	end
end

File: csharp/gmsec4/dox/field/Field.cs
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>
/// Base class for all field types.
/// </summary>

public class Field : global::System.IDisposable {

  /// <summary>Returns the name of the field object</summary>
  /// <returns>A string containing the field name</returns>
  public string GetName() {
  }

  /// <summary>Sets the name of the Field object</summary>
  /// <param name="name">The name of the Field object.</param>
  /// <exception cref="GmsecException">An exception is thrown if the name is null or contains an empty string.</exception>
  public void SetName(string name) {
  }

  /// <summary>Returns the type of the field object</summary>
  /// <returns>The field type</returns>
  public Field.FieldType GetFieldType() {
  }

  /// <summary>Returns whether the field object is considered a header field</summary>
  /// <returns>True if a header field; false otherwise</returns>
  public bool IsHeader() {
  }

  /// <summary>Can be used to indicate whether the Field object is a header field</summary>
  /// <param name="header">True if a header field; false otherwise</param>
  public void IsHeader(bool header) {
  }

  /// <summary>Converts the field object into an XML string representation</summary>
  /// <returns>An XML string</returns>
  public virtual string ToXML() {
  }

  /// <summary>Converts the field object into a JSON string representation</summary>
  /// <returns>A JSON string</returns>
  public virtual string ToJSON() {
  }

  /// <summary>Attempts to convert the field value into a signed 64-bit integer number representation.</summary>
  /// <returns>Returns the field value as a signed integer.</returns>
  /// <exception cref="GmsecException">An exception is thrown if the field cannot be successfully converted to an integer.</exception>
  public long GetIntegerValue() {
  }

  /// <summary>Attempts to convert the field value into an unsigned 64-bit integer number representation.</summary>
  /// <returns>Returns the field value as an unsigned integer.</returns>
  /// <exception cref="GmsecException">An exception is thrown if the field cannot be successfully converted to an integer.</exception>
  public ulong GetUnsignedIntegerValue() {
  }

  /// <summary>Attempts to convert the field value into a 64-bit floating point number representation.</summary>
  /// <returns>Returns the field value as a floating point number.</returns>
  /// <exception cref="GmsecException">An exception is thrown if the field cannot be successfully converted to a double.</exception>
  public double GetDoubleValue() {
  }

  /// <summary>Attempts to convert the field value into string representation.</summary>
  /// <returns>Returns the field value as a string.</returns>
  /// <exception cref="GmsecException">An exception is thrown if the field cannot be successfully converted to a string.</exception>
  public string GetStringValue() {
  }

  /// <summary>Creates/returns a clone of the Field object.</summary>
  /// <returns>A clone of the Field object.</returns>
  public Field Clone() {
  }

  /// <summary>Destroys the cloned Field object.</summary>
  /// <param name="clone">The cloned Field object to destroy.</param>
  public void DestroyClone(Field clone) {
  }

  /// <summary>Enumeration of possible field types.</summary>
  public enum FieldType {
    /// <summary>Field containing a character value.</summary>
    CHAR_TYPE = 1,

    /// <summary>Field containing a boolean value.</summary>
    BOOL_TYPE = 2,

    /// <summary>Field containing a signed 16-bit value.</summary>
    I16_TYPE = 3,

    /// <summary>Field containing an unsigned 16-bit value.</summary>
    U16_TYPE = 4,

    /// <summary>Field containing a signed 32-bit value.</summary>
    I32_TYPE = 5,

    /// <summary>Field containing an unsigned 32-bit value.</summary>
    U32_TYPE = 6,

    /// <summary>Field containing a 32-bit float value.</summary>
    F32_TYPE = 7,

    /// <summary>Field containing a 64-bit float value.</summary>
    F64_TYPE = 8,

    /// <summary>Field containing a string value.</summary>
    STRING_TYPE = 9,

    /// <summary>Field containing raw data.</summary>
    BIN_TYPE = 10,

    /// <summary>Field containing a signed 8-bit value.</summary>
    I8_TYPE = 20,

    /// <summary>Field containing an unsigned 8-bit value.</summary>
    U8_TYPE = 21,

    /// <summary>Field containing a signed 64-bit value.</summary>
    I64_TYPE = 22,

    /// <summary>Field containing an unsigned 64-bit value.</summary>
    U64_TYPE = 23
  }

}

}
